omise/libraries/omise-plugin/helpers/charge.php

```diff
@@ -2,25 +2,73 @@
 if (! class_exists('OmisePluginHelperCharge')) {
     class OmisePluginHelperCharge
     {
+
+        private static
+            $currencyDecimals = [
+                "BIF" => 0,
+                "CLP" => 0,
+                "DJF" => 0,
+                "GNF" => 0,
+                "ISK" => 0,
+                "JPY" => 0,
+                "KMF" => 0,
+                "KRW" => 0,
+                "PYG" => 0,
+                "RWF" => 0,
+                "UGX" => 0,
+                "UYI" => 0,
+                "VND" => 0,
+                "VUV" => 0,
+                "XAF" => 0,
+                "XOF" => 0,
+                "XPF" => 0,
+                "BHD" => 3,
+                "IQD" => 3,
+                "JOD" => 3,
+                "KWD" => 3,
+                "LYD" => 3,
+                "OMR" => 3,
+                "TND" => 3,
+            ],
+            $defaultCurrencyDecimals = 2,
+            $upportedCurrencies = [
+                'AUD',
+                'CAD',
+                'CHF',
+                'CNY',
+                'DKK',
+                'EUR',
+                'GBP',
+                'HKD',
+                'JPY',
+                'MYR',
+                'THB',
+                'SGD',
+                'USD'
+            ]
+        ;
+
+        /**
+         * Return number of subunits the passed currency has
+         *
+         * @param string $currency
+         * @return integer
+         */
+        public static function subUnitsFor($curr) {
+            $c = strtoupper($curr);
+            $decimals = isset(self::$currencyDecimals[$c]) ? self::$currencyDecimals[$c] : self::$defaultCurrencyDecimals;
+            return pow(10, $decimals);
+        }        
+
         /**
          *
          * @param string $currency
-         * @param integer $amount
```

Comment:
we could correct comment maybe `@param float $amount` ?

Reply:
It accepts any var that can be 'used' as a number - I guess that would be mainly int/float/numeric string - maybe should just write `@param numeric $amount`